***Dans ce notebook , on essaiera d'analyser les données textuelles***

***Maix avant quelques indices qui peuvent aider dans notre analyse***

In [5]:
import pandas as pd

# Chemin du fichier Excel
file_path = 'space.xlsx'

# Chargement du fichier Excel
df = pd.read_excel(file_path)

# Nombre total d'entrées dans la colonne "Nom" (y compris les répétitions)
total_entries = df['nom'].count()

# Nombre total d'artistes uniques
unique_artists = df['nom'].nunique()

# Affichage des résultats
print(f"Nombre total d'entrées (y compris les répétitions) : {total_entries}")
print(f"Nombre total d'artistes uniques : {unique_artists}")


Nombre total d'entrées (y compris les répétitions) : 332
Nombre total d'artistes uniques : 332


- On peut constater qu'il existe dans ce document 332 artistes

***L'ajout des colonnes "Question" , "Réponse"***

In [6]:

# Ajouter les colonnes "Question" et "Reponse" avec des valeurs par défaut (vides)
df['Question'] = ""  # Ajout d'une colonne vide pour les questions
df['Reponse'] = ""   # Ajout d'une colonne vide pour les réponses

# Sauvegarder le fichier Excel mis à jour
df.to_excel(file_path, index=False)

print("Les colonnes 'Question' et 'Reponse' ont été ajoutées au fichier Excel.")

Les colonnes 'Question' et 'Reponse' ont été ajoutées au fichier Excel.


In [8]:
print(df.columns)

Index(['id', 'nom', 'date_ouverture', 'date_fermeture', 'website', 'email',
       'adresse', 'horaires', 'presentation', 'responsables', 'historique',
       'activites', 'expositions', 'publications', 'events', 'residences',
       'workshop', 'archive', 'librairie', 'coworking', 'education',
       'nb_expos_an', 'nb_artistes_an', 'partenaires', 'created_at',
       'updated_at', 'slug', 'pays', 'ville', 'latitude', 'longitude',
       'adresse_complete', 'codepostal', 'telephone', 'visite', 'user_id',
       'Question', 'Reponse'],
      dtype='object')


***Extraction des noms des espaces***

In [52]:
import pdfplumber
import re
from collections import Counter

# Chemin vers le fichier PDF
file_path = 'qstn_rpns.pdf'

# Liste pour stocker les noms des artistes/espaces
artist_names = []

# Ouverture du fichier PDF
with pdfplumber.open(file_path) as pdf:
    for page in pdf.pages:
        text = page.extract_text()
        if text:
            # Expression régulière mise à jour pour exclure les cas précédés de "de" ou suivis de texte contextuel
            matches = re.findall(r"(?<![a-zA-Z0-9,(\s])([A-Za-z0-9\s'’\-\.]+)\s*\(\d{4}\s*-\s*(?:\d{4})?\)(?!\s*de\s*[A-Z])", text)
            # Ajouter les résultats à la liste des noms, en nettoyant les espaces supplémentaires
            artist_names.extend([match.strip() for match in matches if len(match.strip()) > 1])

# Compter le nombre d'occurrences de chaque artiste/nom d'espace
artist_count = Counter(artist_names)

# Afficher les noms des artistes/espaces et leur nombre d'occurrences
for artist, count in artist_count.items():
    print(f"{artist}: {count}")


IMMANENCE: 1
picerie: 1
Donnant-Donnant: 1
L’espace d’en bas: 1
ditions de
l’espace d’en bas: 1
Collection adresse: 1
PAMELA
artist-run space: 1
PeepSpace: 1
Kurt-forever: 1
6b: 1
DISPLAY: 1
Palais des paris: 1
Lieu-Commun: 1
GLASSBOX: 1
artist-run-spaces.org: 1
CHOTONE: 1
Poteaux d’angle: 1
Red Cactus Art
Studio: 1
GLASSBOX-NORD: 1
GLASSBOX-SUD: 1
DOS MARES: 1


- Il reste toujours des problémes de lecture , par exemple dans la sortie de code, l'un des noms "picerie" alors que c'est "l'épierie"

In [55]:
import pandas as pd

# Chemin vers le fichier Excel (utilisez un chemin absolu si nécessaire)
file_path = 'space.xlsx'

# Liste des valeurs à vérifier dans la colonne "Nom"
values_to_check = [
    "IMMANENCE", "L'épicerie", "Donnant-Donnant", "L’espace d’en bas", 
    "ditions de l’espace d’en bas", "Collection adresse", "PAMELA",
    "artist-run space", "PeepSpace", "Kurt-forever", "6b", "DISPLAY",
    "Palais des paris", "Lieu-Commun", "GLASSBOX", "artist-run-spaces.org",
    "ECHOTONE", "Poteaux d’angle", "Red Cactus Art Studio", "GLASSBOX-NORD",
    "GLASSBOX-SUD", "DOS MARES"
]

# Chargement du fichier Excel
df = pd.read_excel(file_path)

# Vérifier si les valeurs existent dans la colonne "Nom" et les stocker dans un dictionnaire
values_existence = {value: value in df['nom'].values for value in values_to_check}

# Affichage des résultats
for name, exists in values_existence.items():
    print(f"{name}: {'Présent' if exists else 'Absent'}")


IMMANENCE: Absent
L'épicerie: Absent
Donnant-Donnant: Absent
L’espace d’en bas: Absent
ditions de l’espace d’en bas: Absent
Collection adresse: Absent
PAMELA: Absent
artist-run space: Absent
PeepSpace: Absent
Kurt-forever: Absent
6b: Présent
DISPLAY: Présent
Palais des paris: Absent
Lieu-Commun: Absent
GLASSBOX: Absent
artist-run-spaces.org: Absent
ECHOTONE: Absent
Poteaux d’angle: Absent
Red Cactus Art Studio: Présent
GLASSBOX-NORD: Absent
GLASSBOX-SUD: Absent
DOS MARES: Absent


*  On peut remarquer que la plupart des noms n'existent pas dans notre jeux de données et cela peut etre a cause de : 
-  L'orthogrphe
- L'etude n'est pas encore finie 